src/main/java/teammates/ui/controller/InstructorCourseDetailsPageData.java

```diff
@@ -8,23 +8,165 @@
 import teammates.common.datatransfer.StudentAttributes;
 import teammates.common.util.Const;
 import teammates.common.util.Url;
+import teammates.ui.template.CourseDetailsStudentsTable;
+import teammates.ui.template.CourseDetailsStudentsTableRow;
+import teammates.ui.template.ElementTag;
 
 /**
  * PageData: data used for the "Course Details" page
  */
 public class InstructorCourseDetailsPageData extends PageData {
+    private InstructorAttributes currentInstructor;
+    private CourseDetailsBundle courseDetails;
+    private List<InstructorAttributes> instructors;
+    private String studentListHtmlTableAsString;
+    private ElementTag giveCommentButton;
+    private ElementTag courseRemindButton;
+    private CourseDetailsStudentsTable studentsTable;
     
     public InstructorCourseDetailsPageData(AccountAttributes account) {
         super(account);
     }
+    
+    public void init(InstructorAttributes currentInstructor, CourseDetailsBundle courseDetails, 
+                     List<InstructorAttributes> instructors, List<StudentAttributes> students) {
+        this.currentInstructor = currentInstructor;
+        this.courseDetails = courseDetails;
+        this.instructors = instructors;
+        
+        boolean isDisabled = !currentInstructor.isAllowedForPrivilege(
+                                                    Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+        
+        String content = "<span class=\"glyphicon glyphicon-comment glyphicon-primary\"></span>";
+        giveCommentButton = createButton(content, "btn btn-default btn-xs icon-button pull-right", 
+                                         "button_add_comment", null, "", "tooltip", null, isDisabled);
+        
+        isDisabled = !currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+        String onClick = "if(toggleSendRegistrationKeysConfirmation('" 
+                          + sanitizeForJs(courseDetails.course.id) + "')) "
+                          + "window.location.href='" + sanitizeForJs(getInstructorCourseRemindLink()) + "';";
+        courseRemindButton = createButton(null, "btn btn-primary", "button_remind", null, 
+                                          Const.Tooltips.COURSE_REMIND, "tooltip", onClick, isDisabled);
+        
+        studentsTable = new CourseDetailsStudentsTable();
+        int studentIndex = 0;
+        for (StudentAttributes student : students) {
+            CourseDetailsStudentsTableRow row = createStudentsTableRow(studentIndex, student);
+            studentsTable.getRows().add(row);
+            studentIndex++;
```

Comment:
Likewise, maybe create the list before passing it into `CourseDetailsStudentsTable` here as well.